pyroengine/engine/engine.py

```diff
@@ -13,44 +13,127 @@ class PyronearEngine:
+                pi_zero_id = '-1'  # add default key value
+
+            if not self.event_appening[pi_zero_id]:
+                self.detection_counter[pi_zero_id] += 1
+                # Ensure counter max value is 3
+                if self.detection_counter[pi_zero_id] > 3:
+                    self.detection_counter[pi_zero_id] = 3
+
+            # If counter reach 3, start sending alerts
+            if self.detection_counter[pi_zero_id] == 3:
+                self.event_appening[pi_zero_id] = True
+
+            if self.use_api and self.event_appening[pi_zero_id]:
+                frame.save(self.stream, format='JPEG')
+                # Send alert to the api
+                self.send_alert(pi_zero_id)
+                self.stream.seek(0)  # "Rewind" the stream to the beginning so we can read its content
+
+        else:
+            if self.detection_counter[pi_zero_id] > 0:
+                self.detection_counter[pi_zero_id] -= 1
+
+            if self.detection_counter[pi_zero_id] == 0 and self.event_appening[pi_zero_id]:
+                # Stop event
+                self.event_appening[pi_zero_id] = False
+
+        # save frame
+        if self.use_api and self.save_evry_n_frame:
+            self.frames_counter[pi_zero_id] += 1
+            if self.frames_counter[pi_zero_id] == self.save_evry_n_frame:
+                # Reset frame counter
+                self.frames_counter[pi_zero_id] = 0
+                # Send frame to the api
+                frame.save(self.stream, format='JPEG')
+                self.save_frame(pi_zero_id)
+                self.stream.seek(0)  # "Rewind" the stream to the beginning so we can read its content
 
         return res
 
-    def init_api(self, api_login, api_password):
+    def init_api(self, pi_zero_credentials):
         """Setup api"""
-        self.api_client = client.Client(self.api_url, api_login, api_password)
-        self.event_id = self.api_client.create_event(lat=9, lon=9).json()["id"]
+        self.api_client = {}
+        for pi_zero_id in pi_zero_credentials.keys():
+            self.api_client[pi_zero_id] = client.Client(self.api_url, pi_zero_credentials[pi_zero_id]['login'],
+                                                        pi_zero_credentials[pi_zero_id]['password'])
 
-    def send_alert(self):
-        """Send Alert"""
+    def send_alert(self, pi_zero_id):
+        """Send alert"""
+        print("Send alert !")
         # Create a media
-        media_id = self.api_client.create_media_from_device().json()["id"]
+        media_id = self.api_client[pi_zero_id].create_media_from_device().json()["id"]
         # Create an alert linked to the media and the event
-        self.api_client.send_alert_from_device(lat=9, lon=9, event_id=self.event_id, media_id=media_id)
-        self.api_client.upload_media(media_id=media_id, image_data=self.image.getvalue())
+        self.api_client[pi_zero_id].send_alert_from_device(lat=9, lon=9, media_id=media_id)
```

Comment:
`lat`, `lon` will be retrieved by the device API `lat` `lon` rather than config data ? (as in [here](http://pyronear-api.herokuapp.com/redoc#operation/register_my_device_devices_register_post))

Reply:
done !

Reply:
Thanks, could you please also add `azimuth`

Reply:
according to @frgfm azimuth should be fix for each device, therefore it's not managed here

Reply:
Same as `lat` `lon` if I m correct ? I do not see the point of letting this empty as send_alert_from_device() does take it as param ...

Reply:
yes, for me they should not be here either but I am obliged to pass them because the api is designed like that, I did not really understand the logic. @frgfm maybe you can enlighten us

Reply:
Originally we were suppose to send the actual `lat` `lon` of the fire itself that is why I guess this was meant to be passed with `send_alert_from_device()`.